src/pages/PageLayout.tsx

```diff
@@ -1,39 +1,124 @@
-import { Container, ContainerProps } from "@threshold-network/components"
-import { Outlet } from "react-router-dom"
-import SubNavigationPills from "../components/SubNavigationPills"
-import { PageComponent } from "../types"
-import useDocumentTitle from "../hooks/useDocumentTitle"
-
-export interface PageLayoutProps extends ContainerProps {
-  pages?: PageComponent[]
-  title?: string
+import {
+  Box,
+  Flex,
+  FlexProps,
+  Grid,
+  SystemStyleObject,
+} from "@threshold-network/components"
+import { FC, ReactNode } from "react"
+import PRIMARY_BACKGROUND_PATH from "../static/images/layout-background-primary.svg"
+import SECONDARY_BACKGROUND_PATH from "../static/images/layout-background-secondary.svg"
+
+const HORIZONTAL_PADDING = "13.625rem" // 218px
+const BORDER = "1px solid hsla(0, 0%, 100%, 10%)"
+
+//TODO: Remove stale layout components eg. `./Bridge/BridgeLayout.tsx` etc...
+
+type VariantType = "primary" | "secondary"
+type RenderSlotType = Nullable<ReactNode>
+
+interface PageLayoutProps {
+  /**
+   * Render slot for content aligned at the top.
+   *
+   * @default undefined
+   */
+  renderTop?: RenderSlotType
+  /**
+   * Render slot for content aligned on the left. If the value is `null` the
+   * area is left empty. If the value is not provided the main content (children
+   * occupies the area.
+   *
+   * @default undefined
+   */
+  renderLeft?: RenderSlotType
+  /**
+   * Render slot for content aligned on the right. If the value is `null` the
+   * area is left empty. If the value is not provided the main content (children)
+   * occupies the area.
+   *
+   * @default undefined
+   */
+  renderRight?: RenderSlotType
+  /**
+   * The variant of the background.
+   *
+   * @default "primary"
+   *  */
+  backgroundVariant?: VariantType
+  /** The main content of the page. */
+  children: ReactNode
 }
 
-const PageLayout: FC<PageLayoutProps> = ({
-  pages,
-  title,
-  children,
-  ...restProps
-}) => {
-  useDocumentTitle(`Threshold - ${title}`)
-  const links = pages
-    ? pages.filter((_) => !_.route.hideFromMenu).map((_) => _.route)
-    : []
+const getBackgroundStyles = (variant: VariantType): SystemStyleObject => ({
+  bgImage:
+    variant === "primary" ? PRIMARY_BACKGROUND_PATH : SECONDARY_BACKGROUND_PATH,
+  bgRepeat: "no-repeat",
+  bgSize: variant === "primary" ? "contain" : "cover",
+  bgPosition: "bottom",
+  bgAttachment: "fixed",
+})
+
+const FixedContainer: FC<FlexProps> = (props) => (
+  <Flex flexFlow={"column"} maxW={"1920px"} mx={"auto"} {...props} />
+)
+
+/**
+ * Renders the page wrapped in layout with render slots for content alignment.
+ * @param {PageLayoutProps} props - The props for the PageLayout component.
+ * @return {JSX.Element} The rendered page wrapped in layout.
+ */
+const PageLayout: FC<PageLayoutProps> = (props) => {
+  const {
+    renderTop,
+    renderLeft,
+    renderRight,
+    backgroundVariant = "primary",
+    children,
+    ...restProps
+  } = props
```

Comment:
I'm not sure what was the point to refactor the `PageLayout` component tbh 🤔  I also don't think implementing it like this is a good idea.

The `PageLayout` should be reusable for every page and it should contain only necessary and very basic thing, that applies for every page. This is why we've only had `Container` component there as it's common for all the pages. The actual implementation assumes that the page will contain the maximum of 3 column.

The second thing is, that this implementation is not responsive at all. Decreasing the width of the screen decreases the width of the 'Outlet' component instead of reducing the margins on the sides.

<img width="946" alt="image" src="https://github.com/threshold-network/bitcoin-on-base/assets/40306834/6c6591d6-1f86-4c0a-b895-d720633b6f51">

Next thing, putting slots to render as props is also a bad idea imo. The `PageLayout` should be more "fluid" and we shouldn't assume that this will be reusable for all pages. For the `Minting` flow we should create a separate layout and name it differently. I think it will be anough if we create one horizontal stack (that is only visible when user is connected) and three rows. We can also use Grid for that. For the Bridge main page we should implement a separate layout that only contains two columns that are responsive. I believe the Bridge main page is not implemented in this PR so let's focus on minting layout first.

Reply:
> The `PageLayout` should be reusable for every page 

Actually it is. It covers every scenario presented in designs and can be freely used.

> This is why we've only had `Container` component

The `PageLayout` now acts as this component. My objective was to get rid of "pseudo" reusable components like `<Something>Layout` (+ confusing wrapper component somewhere in between) in favor of truely reusable, multipurpose `PageLayout` component.

I have implemented responsive styles. It behaves (almost) like Threshold Dashboard. Ref commit: 9a864bd2cdfc53d6095a07f97f6bba0fff5d2177

Now `Header` component from #13 might need some fine tuning regarding to alignment between menu and content but I'm not sure how it should behave. I'd leave as is for now. What do you think?

Reply:
Regarding the implementation, let's move the discussion here: https://github.com/threshold-network/bitcoin-on-base/pull/15#discussion_r1467402522

Regarding `Header` component and it's behaviour - I think we can leave it for now. We can improve it later if needed when we know more about how it should behave.